classes/author/maps/Schema.inc.php

```diff
@@ -78,7 +79,26 @@ protected function mapByProperties(array $props, Author $item): array
     {
         $output = [];
         foreach ($props as $prop) {
-            $output[$prop] = $item->getData($prop);
+            switch ($prop) {
+                case 'userGroupLabel':
+                    $userGroupDao = \DAORegistry::getDAO('UserGroupDAO'); /** @var \UserGroupDAO $userGroupDao */
+                    $authorUserGroups = $userGroupDao->getByRoleId($this->context->getId(), Role::ROLE_ID_AUTHOR)->toArray();
+
+                    $currentAuthorUserGroupFiltered = array_filter($authorUserGroups, function ($authorUserGroup) use ($item) {
+                        return $authorUserGroup->getId() == $item->getData('userGroupId');
+                    });
+
+                    if (!empty($currentAuthorUserGroupFiltered)) {
+                        $currentAuthorUserGroup = array_values($currentAuthorUserGroupFiltered)[0];
+
+                        $output[$prop] = $currentAuthorUserGroup->getLocalizedName();
+                    }
+
+                    break;
```

Comment:
You can use the map class's constructor to pre-fetch data from the database _once_ and use it when mapping a whole collection. That would look like this:

```php
class Schema extends \PKP\core\maps\Schema
{
    public array $userGroupLabels;

    public function __construct(PKPRequest $request, Context $context, PKPSchemaService $schemaService)
    {
        parent::__construct($request, $context, $schemaservice);

        $userGroupDao = \DAORegistry::getDAO('UserGroupDAO'); /** @var \UserGroupDAO $userGroupDao */
        $authorUserGroups = $userGroupDao->getByRoleId($this->context->getId(), Role::ROLE_ID_AUTHOR)->toArray();
        foreach (...) {
            $this->userGroupLabels[$id] => $label;
        }
    }

    protected function mapByProperties(array $props, Author $item): array
    {
        $output = [];
        foreach ($props as $prop) {
            switch ($prop) {
                case 'userGroupLabel':
                    $output[$prop] = $this->userGroupLabels[$item->getData('userGroupId')];

                    break;
                
        // ...
    }
}
```

Reply:
Nice! Fixed.